src/core/dep-graph.ts

```diff
@@ -138,6 +138,38 @@ class DepGraphImpl implements types.DepGraphInternal {
     return count;
   }
 
+  public isSame(other: types.DepGraph): boolean {
+    const sortByNodeId = (nodeA: any, nodeB: any) => nodeA.nodeId.localeCompare(nodeB.nodeId);
+
+    const selfNodes = this.toJSON().graph.nodes.sort(sortByNodeId);
```

Comment:
you don't have to do this, you can cast `other` to `DepGraphImpl` and compare the privates ... maybe it's better to avoid some wasted CPU cycles?

Reply:
I agree, but `DepGraph` is the only public interface here. I can't be sure `other` is the `DepGraphImpl` implementation.

Reply:
I think you can assume it. We already do something similar in https://github.com/snyk/dep-graph/blob/master/src/legacy/index.ts#L203

Reply:
I don't think this is a great idea, its ok in "legacy conversion" as we want that out as fast as possible, but we can't guarantee that the runtime types are what we expect can we?

Reply:
if we need to do this for efficiency reasons than maybe that's telling us something about the interface itself? TS is not JS

Reply:
@gjvis you have a point. In theory 2 different versions of the lib can exist in the same app and 2 incompatible instances can "meet" in this comparison function.

Doing 
```typescriprt
fromJSON(other.toJSON()) 
```
shouldn't be expensive, and is indeed safer.

and yep it might hint of a better potential interface 🤔 ... can't see it right now.